packages/studio/src/components/AddPageButton/StreamScopeCollector.tsx

```diff
@@ -1,58 +1,185 @@
+import { StreamScope } from "@yext/studio-plugin";
+import PillPickerInput from "../PillPicker/PillPickerInput";
+import { streamScopeFormData } from "../PageSettingsButton/EntityPageModal";
 
 export default function StreamScopeCollector({
   isOpen,
   handleClose,
   handleConfirm,
 }: FlowStepModalProps) {
-  const { actions } = useContext(AddPageContext);
-
-  const onConfirm = useCallback(
-    async (data: StreamScopeForm) => {
-      const streamScope = StreamScopeParser.parseStreamScope(data);
-      actions.setStreamScope(streamScope);
-      await handleConfirm();
-      return true;
+  const { state, actions } = useContext(AddPageContext);
+  const { streamScope } = state;
+  const { setStreamScope } = actions;
+  const [savedFilters, entitiesRecord] = useStudioStore((store) => [
+    store.accountContent.savedFilters,
+    store.accountContent.entitiesRecord,
+  ]);
+
+  useEffect(() => {
+    if (streamScope === undefined) {
+      setStreamScope({});
+    }
+  }, [setStreamScope, streamScope]);
+
+  const updateScopeItems = useCallback(
+    (field: string) => (selectedItems: string[] | undefined) => {
+      const updatedStreamScope = {
+        ...streamScope,
+        [field]: selectedItems,
+      };
+      if (!selectedItems?.length) {
+        delete updatedStreamScope[field];
+      }
+      setStreamScope(updatedStreamScope);
     },
-    [handleConfirm, actions]
+    [streamScope, setStreamScope]
+  );
+
+  const scopeFormData: {
+    [field in keyof StreamScope]: Pick<
+      ScopeFieldProps,
+      "emptyText" | "allItems" | "getDisplayName"
+    >;
+  } = useMemo(
+    () => ({
+      entityIds: {
+        emptyText: "No entities found in the account.",
+        // TODO (SLAP-2907): Populate dropdown from store
+        allItems: [],
+      },
+      entityTypes: {
+        emptyText: "No entity types found in the account.",
+        allItems: Object.keys(entitiesRecord),
+      },
+      savedFilterIds: {
+        emptyText: "No saved filters found in the account.",
+        allItems: savedFilters.map((f) => f.id),
+        getDisplayName: (item) => {
+          const savedFilter = savedFilters.find((f) => f.id === item);
+          return savedFilter ? getDisplayString(savedFilter) : item;
+        },
+      },
+    }),
+    [savedFilters, entitiesRecord]
   );
 
+  const modalBodyContent = useMemo(() => {
+    const totalStreamScopeItems = Object.values(streamScope ?? {}).reduce(
+      (numItems, scopeItems) => {
+        return numItems + scopeItems.length;
+      },
+      0
+    );
+
+    return (
+      <>
+        <div className="italic mb-4">
+          Use one of the optional fields below to specify which entities this
+          page can access.
+        </div>
+        {Object.entries(scopeFormData).map(([field, data]) => {
+          const selectedItems: string[] | undefined = streamScope?.[field];
+          const hasOtherScopeFilters =
+            totalStreamScopeItems > (selectedItems?.length ?? 0);
```

Comment:
To determine if there are other Scope Filters, could we just check if `streamScope` is non-empty?

Reply:
No, that would disable all `ScopeField`s once a single filter was added to the stream scope. We only want to check if other scope types have filters (i.e. there are more filters for the scope beyond the ones for this particular "type") so users can continue adding filters of the same type

Reply:
Gotcha!